homeassistant/helpers/condition.py

```diff
@@ -112,15 +112,22 @@ def condition_trace_update_result(**kwargs: Any) -> None:
 @contextmanager
 def trace_condition(variables: TemplateVarsType) -> Generator:
     """Trace condition evaluation."""
-    trace_element = condition_trace_append(variables, trace_path_get())
-    trace_stack_push(trace_stack_cv, trace_element)
+    should_pop = True
+    trace_element = trace_stack_top(trace_stack_cv)
+    if trace_element and trace_element.reuse_by_child:
+        should_pop = False
+        trace_element.reuse_by_child = False
+    else:
+        trace_element = condition_trace_append(variables, trace_path_get())
```

Comment:
So now it either reuses the top child or it creates a new one under `condition`. Why not always expect the caller to have set the right path ? Because the trace condition can't make up paths because it doesn't know the location it has been defined

Reply:
The code already relies on the caller setting up the path, before and after this change.

This change is about reusing an existing trace element for the root node of a condition.

Reply:
Right. But why don't we always reuse the existing trace element since the `trace_condition` function can't know what to add to the path to make it point at the condition config?